libraries/botbuilder-azure-blobs/src/blobsTranscriptStore.ts

```diff
@@ -32,14 +32,14 @@ function getConversationPrefix(channelId: string, conversationId: string): strin
     return sanitizeBlobKey(`${channelId}/${conversationId}`);
 }
 
+const DateT = t.Guard((val: unknown): val is Date => val instanceof Date, { name: 'Date' });
+
 // Formats an activity as a blob key
 function getBlobKey(activity: Activity): string {
-    assert.date(activity.timestamp, ['activity', 'timestamp']);
+    const { timestamp } = t.Record({ timestamp: DateT }).check(activity);
 
     return sanitizeBlobKey(
-        [activity.channelId, activity.conversation.id, `${formatTicks(activity.timestamp)}-${activity.id}.json`].join(
-            '/'
-        )
+        [activity.channelId, activity.conversation.id, `${formatTicks(timestamp)}-${activity.id}.json`].join('/')
```

Comment:
Not something for this PR, but in v5 the behavior around generating keys needs to be locked down and inputs need to be validated.

Are `null`, `undefined`, empty strings or whitespace-only strings valid for conversation IDs?